### DISCUSSION TITLE
What is the way you close the ChatGPT Desktop application?

### DISCUSSION THREAD

**Discussion by D3vil0p3r** [lencx/ChatGPT#205] [Polls]
Hey guys!
Me and @lencx are doing some technical analysis and we would like to ask to you in which way you close the ChatGPT Desktop application when you finish your work on it.

**Comment by expandpi** [comment]
Yaa, i also saw it and whenever i open the chatGPT from dock or menu it creates one more process everytime and it start taking my system resources. It could be a good idea to make this application completely close with close button.

**Comment by lencx** [comment]
What we need to think about is why applications repeatedly open multiple processes in Linux, instead of simply using the window button to actually close it to solve all the problems, since the application already provides two ways to exit (menu and shortcut). It's hard to balance the user's habits, and I think what needs to be done is to guide the user on how to actually close it, not by modifying the current functionality.

This application is positioned as a tool for high frequency use, and the reason it cannot be closed by the window button is to prevent users from accidentally touching it.

**Reply by D3vil0p3r** [reply]
I think that if 100 applications in an environment close by a close button on window top bar, users would think that all applications will work in this manner. Btw, guessing that you don't want to implement this fix, at least you should fix the issue when, if I open a first time ChatGPT app, then I close its window and it continues to run in background, if I open it a second time, it should reopen the first ChatGPT app process, and not making a second one as happens now.

**Reply by lencx** [reply]
Many resident apps don't support direct exit, they often exist in the system tray. They make sense to me, at least as far as my understanding of app interaction goes.

I'm not sure what the point of the 100 apps you mentioned is? Are they all positioned as high frequency use software? If it's not high frequency use, I don't think there's a need to use the app as it will make more features in the future to meet its true high frequency positioning.

**Reply by D3vil0p3r** [reply]
> Many resident apps don't support direct exit, they often exist in the system tray. They make sense to me, at least as far as my understanding of app interaction goes.

I think that in a usual view, having an running application that, when u close its windows (so it is running in background) open it N times, it creates new N processes, it is not normal... The normal approach at least should be that, if you have a running application, and you click on its icon, instead of opening N new application processes, keep ahead only the running one... I hope that at least with this you agree, because having N running process of the same application is not normal...

**Reply by lencx** [reply]
> What we need to think about is why applications repeatedly open multiple processes in Linux, instead of simply using the window button to actually close it to solve all the problems.

As I already said, it's a problem that needs to be solved, not using a button close to escape the multi-process problem. I'd be happy to see a similar PR. as I'm not sure why the application would have multiple processes on Linux.

**Reply by D3vil0p3r** [reply]
> As I already said, it's a problem that needs to be solved, not using a button close to escape the multi-process problem. I'd be happy to see a similar PR. as I'm not sure why the application would have multiple processes on Linux.

Yes, I agree: if the multi-process issue can be solved, the window "close" button function can remain untouched.

**Reply by D3vil0p3r** [reply]
@lencx I noted that Discord app in Linux manages this multi-process issue. For example, if you open Discord by the terminal a first time, it opens. Then, if you open another terminal, and run again discord, you get the message "Quitting secondary instance." so it means that, they implemented something that, when you run the application, check if there is already a primary instance. If yes, it quits the new secondary instance. I think it is the easiest approach to solve the multi-process issue.

**Reply by lencx** [reply]
I am not familiar with Linux, and I do not currently have a Linux platform to develop and verify.

**Reply by D3vil0p3r** [reply]
> I am not familiar with Linux, and I do not currently have a Linux platform to develop and verify.

For testing other environments different than MacOS (I'm guessing you are a MacOS user according to your answers), you can use virtualization software for virtualizing OSes like Linux and test the apps that you develop.
